From 6a06d06670efff6865961bf0152592a00ce047ac Mon Sep 17 00:00:00 2001
From: Thomas Bourgoin <thomas.bourgoin@foss.st.com>
Date: Mon, 3 Feb 2025 17:37:53 +0100
Subject: [PATCH 1503/1695] drivers: stm32_serc: disable SERC during suspend
 sequence

Set SERFEN to 0 to disable SERC peripheral before going to low power.
SERC does not work during low power sequence.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git f740e13aa2d90b4a7d9d1abf6fe9d4f1e7d07a77 drivers: stm32_serc: disable SERC during suspend sequence]
Change-Id: Ib79c09feedcd71afa9ba0b5e05f532e02e376213
Signed-off-by: Thomas Bourgoin <thomas.bourgoin@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/434008
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
---
 core/drivers/firewall/stm32_serc.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/core/drivers/firewall/stm32_serc.c b/core/drivers/firewall/stm32_serc.c
index 3c4a646b7..f8c327a41 100644
--- a/core/drivers/firewall/stm32_serc.c
+++ b/core/drivers/firewall/stm32_serc.c
@@ -223,6 +223,8 @@ static TEE_Result stm32_serc_pm(enum pm_op op, unsigned int pm_hint,
 		stm32_serc_setup();
 
 	} else {
+		io_clrbits32(serc_dev.pdata.base + _SERC_ENABLE,
+			     _SERC_ENABLE_SERFEN);
 		clk_disable(serc_dev.pdata.clock);
 	}
 
-- 
2.47.2

